### PR TITLE
Unspecified initial states in PEtab conditions table default to SBML initial value

### PR DESCRIPTION
**Pull request by dilpath**
https://github.com/AMICI-dev/AMICI/issues/1396

The warning added to "petab_import.py" is untested, just seems reasonable to me.

The warning added to "petab_objective.py" can result in many warnings, by default.

`float()` is used in `petab_objective.py` since any SBML initial assignment (for states in the PEtab conditions table) would have been overwritten in "petab_import.py" anyway, so I guess those overwritten SBML initial assignments should be ignored everywhere, hence only float-type values (initial amounts or concentrations) would be expected.

Not a nice solution I guess, since a private method from "sbml_import.py" is used.

### CODE REVIEW

**Review by FFroehlich — APPROVED** [review]
Is there an appropriate test case in the petab testsuite?

**Comment by FFroehlich** [comment]
> Not a nice solution I guess, since a private method from "sbml_import.py" is used.

I don't think thats a big issue, feel free to make this a public method. I generally make all functions private by default as this slims the API and makes it easier to navigate.

**Comment by dilpath** [comment]
> Is there an appropriate test case in the petab testsuite?

Not sure. @elbaraim could your PEtab problem from #1396 be added there?

File: python/amici/petab_objective.py [hunk 1]
```diff
@@ -343,6 +344,17 @@ def _set_initial_concentration(condition_id, species_id, init_par_id,
                                        par_map, scale_map):
             value = petab.to_float_if_float(
                 petab_problem.condition_df.loc[condition_id, species_id])
+            if pd.isna(value):
+                value = float(
+                    get_species_initial(
+                        petab_problem.sbml_model.getSpecies(species_id)
+                    )
+                )
+                logger.warning(f'The species {species_id} has no initial value'
```
Comment (dweindl) [inline python/amici/petab_objective.py:353]: Not sure I'd go for warnings, as this is explicitly a feature in PEtab. Might be confusing. Maybe change to info?

Comment (dilpath) [reply python/amici/petab_objective.py]: Ah, maybe not worth logging then. Switched to debug but could also remove.

Comment (dweindl) [reply python/amici/petab_objective.py]: I think debug can be helpful

File: python/amici/petab_import.py [hunk 2]
```diff
@@ -532,6 +532,13 @@ def import_model_sbml(
         if assignment is None:
             assignment = sbml_model.createInitialAssignment()
             assignment.setSymbol(assignee_id)
+        else:
+            logger.warning('The SBML model has an initial assignment defined '
```
Comment (dweindl) [inline python/amici/petab_import.py:536]: same for this one, would also prefer info/debug

**Review by dweindl — APPROVED** [review]

**Comment by elbaraim** [comment]
> > Is there an appropriate test case in the petab testsuite?
> 
> Not sure. @elbaraim could your PEtab problem from #1396 be added there?

The PEtab example I provided in the issue? Sure. Although I think an even easier model could be used with the same characteristics as in that one. What do you think is more appropriate?

**Comment by elbaraim** [comment]
Hi @dilpath @dweindl , 
There is still an open issue in this PR:
If a preequilibration is defined and the initial state is left empty (or NaN) in the condition table, one would expect that the value that would be taken as initial would be the one obtained from the preequilibration and not from the SBML model.

For debugging: In the example I provided in #1396 one can see this unwanted behavior happening. The condition `unperturbed` is used as preequilibration condition but then the initial value (after preequilibration) is set to the initial value in the SBML (which is zero) instead of the last value in `sol['rdatas'].x_ss`.

```
print('before reinitialization (preeq)')
print(sol['rdatas'][0].x_ss[15])
print(sol['rdatas'][1].x_ss[15])
```
447796069.1255652
447796069.1255652
```
print('after reinitialization')
print(sol['rdatas'][0].x0[15])
print(sol['rdatas'][1].x0[15])
```
2.0
0.0

For the conditions left empty (in this case `unperturbed`) I would expect to get the same value in `x_ss` and `x0` (not zero in this case).
- Is this train of thoughts correct? Maybe I am just wrong 😅

**Comment by dweindl** [comment]
>  If a preequilibration is defined and the initial state is left empty (or NaN) in the condition table, one would expect that the value that would be taken as initial would be the one obtained from the preequilibration and not from the SBML model.

Right. The handling of species in the condition table needs a makeover. There is meanwhile a better test case in the petab test suite and I started fixing stuff (#1344). The problem is, that #1345 needs to be addressed first to fully support those PEtab cases there. This is still some work. But we should at least raise some exception in case things are not fully supported.

**Comment by dweindl** [comment]
Merging as is, even if it does not fully resolve @elbaraim 's issue. The rest will take a bit more time.

**Pull request merged**